homeassistant/components/pterodactyl/sensor.py

```diff
@@ -0,0 +1,195 @@
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfInformation.BYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIGABYTES,
+        suggested_display_precision=1,
+        entity_registry_enabled_default=False,
+    ),
+    PterodactylSensorEntityDescription(
+        key=KEY_NETWORK_OUTBOUND,
+        translation_key=KEY_NETWORK_OUTBOUND,
+        value_fn=lambda data: data.network_outbound,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.DATA_SIZE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfInformation.BYTES,
+        suggested_unit_of_measurement=UnitOfInformation.GIGABYTES,
+        suggested_display_precision=1,
+        entity_registry_enabled_default=False,
+    ),
+    PterodactylSensorEntityDescription(
+        key=KEY_UPTIME,
+        translation_key=KEY_UPTIME,
+        value_fn=lambda data: data.uptime,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        suggested_unit_of_measurement=UnitOfTime.DAYS,
+        suggested_display_precision=1,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: PterodactylConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Pterodactyl sensor platform."""
+    coordinator = config_entry.runtime_data
+
+    async_add_entities(
+        PterodactylSensorEntity(coordinator, identifier, description, config_entry)
+        for identifier in coordinator.api.identifiers
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class PterodactylSensorEntity(PterodactylEntity, SensorEntity):
+    """Representation of a Pterodactyl sensor base entity."""
+
+    entity_description: PterodactylSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PterodactylCoordinator,
+        identifier: str,
+        description: PterodactylSensorEntityDescription,
+        config_entry: PterodactylConfigEntry,
+    ) -> None:
+        """Initialize sensor base entity."""
+        super().__init__(coordinator, identifier, config_entry)
+        self.entity_description = description
+        self._attr_unique_id = f"{self.game_server_data.uuid}_{description.key}"
+        self._update_properties()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._update_properties()
+        self.async_write_ha_state()
+
+    @callback
+    def _update_properties(self) -> None:
+        """Update sensor properties."""
+        self._attr_native_value = self.entity_description.value_fn(
+            self.game_server_data
+        )
```

Comment:
Why don't we use the native value property?

Reply:
I took this implementation from the **minecraft_server** integration. Did the value update handling get easier or did I do it wrong/too complicated in **minecraft_server**? Anyway it looks to be working with just the property `native_value`. At the same time I also added the property `unique_id`, instead of setting the attribute in the constructor.

Reply:
Oh but for unique id we like it in the constructor as its static. There's no wrong or right witht he native_value here, it's just more that a property would be 5 lines as opposed to these 12 and that's a little bit cleaner. I generally prefer the shorthand approach if there's a lot of logic involved with for example climate entities or media players